sdk/eventhub/azure-eventhubs/azure/eventhub/_consumer.py

```diff
@@ -0,0 +1,163 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import unicode_literals
+
+import uuid
+import logging
+
+import uamqp  # type: ignore
+from uamqp import types, errors, utils  # type: ignore
+from uamqp import ReceiveClient, Source  # type: ignore
+
+from .exceptions import _error_handler
+from ._common import EventData, EventPosition
+from ._client_base import ConsumerProducerMixin
+from ._utils import create_properties, trace_link_message
+from ._constants import (
+    EPOCH_SYMBOL,
+    TIMEOUT_SYMBOL,
+    RECEIVER_RUNTIME_METRIC_SYMBOL,
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EventHubConsumer(ConsumerProducerMixin):  # pylint:disable=too-many-instance-attributes
+    """
+    A consumer responsible for reading EventData from a specific Event Hub
+    partition and as a member of a specific consumer group.
+
+    A consumer may be exclusive, which asserts ownership over the partition for the consumer
+    group to ensure that only one consumer from that group is reading the from the partition.
+    These exclusive consumers are sometimes referred to as "Epoch Consumers."
+
+    A consumer may also be non-exclusive, allowing multiple consumers from the same consumer
+    group to be actively reading events from the partition.  These non-exclusive consumers are
+    sometimes referred to as "Non-Epoch Consumers."
+
+    Please use the method `create_consumer` on `EventHubClient` for creating `EventHubConsumer`.
```

Comment:
EventHubClient, EventHubConsumer? I think there is documentation that needs to be scrubbed :)

Reply:
Yes - we will do a separate docs pass. Although docs for this class will no longer be exposed anyway.

Reply:
Yes - we will do a separate docs pass. Although docs for this class will no longer be exposed anyway.